wpilibc/src/main/native/cpp/smartdashboard/Field2d.cpp

```diff
@@ -8,13 +8,15 @@
 #include "frc/smartdashboard/Field2d.h"
 
 #include "frc/smartdashboard/SendableBuilder.h"
+#include "frc/smartdashboard/SendableRegistry.h"
 
 using namespace frc;
 
 Field2d::Field2d() {
   m_objects.emplace_back(
       std::make_unique<FieldObject2d>("Robot", FieldObject2d::private_init{}));
   m_objects[0]->SetPose(Pose2d{});
+  SendableRegistry::GetInstance().Add(this, "Field");
```

Comment:
Unfortunately it's not this easy in C++.  You need to inherit from `SendableHelper<Field2d>` as well to make sure any moves are handled correctly.